nextmv/nextmv/local/geojson_handler.py

```diff
@@ -0,0 +1,323 @@
+
+        # Adjust zoom based on coordinate spread
+        lat_range = max(lats) - min(lats)
+        lon_range = max(lons) - min(lons)
+        max_range = max(lat_range, lon_range)
+
+        if max_range > 10:
+            zoom_level = 4
+        elif max_range > 1:
+            zoom_level = 8
+        elif max_range > 0.1:
+            zoom_level = 10
+        else:
+            zoom_level = default_zoom
+
+        return center_lat, center_lon, zoom_level
+
+    except (KeyError, TypeError, ValueError, IndexError) as e:
+        log(f"Warning: Error calculating map center and zoom from GeoJSON data: {e}")
+        return default_lat, default_lon, default_zoom
+    except Exception as e:
+        log(f"Warning: Unexpected error calculating map center and zoom: {e}")
+        return default_lat, default_lon, default_zoom
+
+
+def extract_geojson_fields(geojson_data: dict) -> tuple[list[str], list[str]]:
+    """
+    Extract available fields for tooltip and popup from GeoJSON data.
+
+    This function analyzes the properties of GeoJSON features to identify
+    suitable fields for displaying in map tooltips and popups. It prioritizes
+    common field names and limits the number of fields to maintain usability.
+
+    Parameters
+    ----------
+    geojson_data : dict
+        A GeoJSON object containing features with properties. Should follow
+        the GeoJSON specification with features containing properties objects.
+
+    Returns
+    -------
+    tuple[list[str], list[str]]
+        A tuple containing (tooltip_fields, popup_fields).
+        - tooltip_fields : list[str]
+            List of field names suitable for tooltips (max 3 fields)
+        - popup_fields : list[str]
+            List of field names suitable for popups (max 5 fields)
+
+    Notes
+    -----
+    - Prioritizes common field names: "name", "title", "label", "id",
+      "popupContent", "description"
+    - Tooltip fields are limited to 3 to prevent overcrowding
+    - Popup fields are limited to 5 to maintain readability
+    - Returns empty lists if no features or properties are found
+    - Gracefully handles malformed GeoJSON data by returning empty lists
+    """
+    tooltip_fields = []
+    popup_fields = []
+
+    try:
+        if "features" not in geojson_data or not geojson_data["features"]:
+            return tooltip_fields, popup_fields
+
+        # Get fields from the first feature's properties
+        first_feature = geojson_data["features"][0]
+        if "properties" not in first_feature or not first_feature["properties"]:
+            return tooltip_fields, popup_fields
+
+        available_fields = list(first_feature["properties"].keys())
+        # Prioritize common field names for tooltip/popup
+        priority_fields = ["name", "title", "label", "id", "popupContent", "description"]
+
+        for field in priority_fields:
+            if field in available_fields:
+                tooltip_fields.append(field)
```

Comment:
I like the addition of tooltips and popups, but I think we should align with what Console does. I think, Console uses specific fields/keys for different intentions:

- "properties.name": Used for filtering (and the popup title?)
- "properties.style": A _map_ that understands "color", "fillColor", etc.
- "properties.metadata": A _list_ of key-value pairs like these: `{"key": "Vehicle ID", "value": "v1"}`

We should sync on this again though. I have an example of this, I think and we can check with David.